### PR TITLE
Adding support for lyra

### PR DESCRIPTION
**Pull request by park-csu**
This pull request adds support for the keyboard lyra.

## Description
This pull requests adds keyboard/layout in `park-csu/lyra` to prevent name collision with already existing keyboard `lyra`

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [x] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* N/A

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/park-csu/lyra/info.json [hunk 1]
```diff
@@ -0,0 +1,118 @@
+{
+    "keyboard_name": "lyra",
+    "url": "",
+    "maintainer": "Changsu Park",
+    "width": 15,
+    "height": 4,
```
Comment (drashna) [inline keyboards/park-csu/lyra/info.json:6]:
```suggestion
```

File: keyboards/park-csu/lyra/keymaps/default/keymap.c [hunk 2]
```diff
@@ -0,0 +1,123 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "config.h"
```
Comment (drashna) [inline keyboards/park-csu/lyra/keymaps/default/keymap.c:18]: This is not needed, as the build process handles this automatically. 
```suggestion
```

File: keyboards/park-csu/lyra/keymaps/default/keymap.c [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+     */
+    [_ADJUST] = LAYOUT( \
+      _______, RESET,   _______, _______, _______, _______,                   _______, _______, _______, _______, _______, KC_DEL, \
+      _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______,                   _______, _______, _______, _______, _______, _______, \
+      _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______, _______ \
+    )
+};
+
+layer_state_t layer_state_set_keymap(layer_state_t state) { return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST); }
```
Comment (drashna) [inline keyboards/park-csu/lyra/keymaps/default/keymap.c:106]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) { return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST); }
```

File: keyboards/park-csu/lyra/keymaps/default/keymap.c [hunk 4]
```diff
@@ -0,0 +1,123 @@
...
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_BRIU);
+        } else {
+            tap_code(KC_BRID);
+        }
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/park-csu/lyra/keymaps/default/keymap.c:123]:
```suggestion
```

File: keyboards/park-csu/lyra/keymaps/park-csu/keymap.c [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include QMK_KEYBOARD_H
+#include "config.h"
```
Comment (drashna) [inline keyboards/park-csu/lyra/keymaps/park-csu/keymap.c:18]:
```suggestion
```

File: keyboards/park-csu/lyra/keymaps/park-csu/keymap.c [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+        }
+    } else if (index == 1) { /* Second encoder */
+        if (clockwise) {
+            tap_code(KC_BRIU);
+        } else {
+            tap_code(KC_BRID);
+        }
+    }
+    return false;
+}
```
Comment (drashna) [inline keyboards/park-csu/lyra/keymaps/park-csu/keymap.c:124]:
```suggestion
layer_state_t layer_state_set_user(layer_state_t state) { return update_tri_layer_state(state, _LOWER, _RAISE, _ADJUST); }

```

File: keyboards/park-csu/lyra/lyra.c [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "lyra.h"
```
Comment (drashna) [inline keyboards/park-csu/lyra/lyra.c:17]: This allows for a default behavior to be defined, without the need for the user to specify the behavior.  This is very useful for the configurator which doesn't have a way to configure this behavior, currently.

```suggestion
#include "lyra.h"

#ifdef ENCODER_ENABLE
bool encoder_update_kb(uint8_t index, bool clockwise) {
    if (!encoder_update_user(index, clockwise)) { return false; }
    if (index == 0) { /* First encoder */
        if (clockwise) {
            tap_code(KC_VOLU);
        } else {
            tap_code(KC_VOLD);
        }
    } else if (index == 1) { /* Second encoder */
        if (clockwise) {
            tap_code(KC_BRIU);
        } else {
            tap_code(KC_BRID);
        }
    }
    return false;
}
#endif
```

File: keyboards/park-csu/lyra/readme.md [hunk 8]
```diff
@@ -0,0 +1,26 @@
+# lyra
+
+![lyra](https://i.imgur.com/DmHl7sz.jpeg)
```
Comment (drashna) [inline keyboards/park-csu/lyra/readme.md:3]:
```suggestion
![lyra](https://i.imgur.com/DmHl7szh.jpeg)
```

File: keyboards/park-csu/lyra/rules.mk [hunk 9]
```diff
@@ -0,0 +1,25 @@
...
+EXTRAKEY_ENABLE = yes       # Audio control and System control
+CONSOLE_ENABLE = no         # Console for debug
+COMMAND_ENABLE = no         # Commands for debug and configuration
+# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
+SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
+RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
+BLUETOOTH_ENABLE = no       # Enable Bluetooth
```
Comment (drashna) [inline keyboards/park-csu/lyra/rules.mk:21]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

**Comment by park-csu** [comment]
@drashna Changed what you requested, can you check it again?

File: keyboards/park-csu/lyra/info.json (hunk 1 above)
Comment (park-csu) [reply keyboards/park-csu/lyra/info.json:6]: This is essencially same right?

**Comment by park-csu** [comment]
My mistake, using the special letter - makes the build impossible. I'll change the name asap

File: keyboards/park-csu/lyra/info.json (hunk 1 above)
Comment (drashna) [reply keyboards/park-csu/lyra/info.json:6]: specifically, these values are not needed, and are automatically generated.  So they're deprecated.

**Review by drashna — CHANGES_REQUESTED** [review]

**Pull request closed without merging**